tests/test_common.py

```diff
@@ -13,53 +13,114 @@ def test_apply_parameter_to_query_do_nothing():
-                '$filter': "title eq '%(title)s'",
-                '$top': "%(top)s"}}
-    parameters = {"title": "the overstory", "top": 3}
-    expected = {'entity': 'books', 'query': {
-                    '$filter': "title eq 'the overstory'",
-                    '$top': 3}}
-    assert nosql_apply_parameters_to_query(query, parameters) == expected
-
-
-@responses.activate
-def test_custom_token_server():
-    auth = Auth(type='custom_token_server',
-                args=['POST', 'https://example.com'],
-                kwargs={'data': {'user': 'u', 'password ': 'p'}, 'filter': '.token'})
-
-    session = auth.get_session()
-    assert isinstance(session.auth, CustomTokenServer)
-
-    responses.add(responses.POST, 'https://example.com', json={'token': 'a'})
-
-    # dummy request class just to introspect headers
-    class TMP:
-        headers = {}
-
-    session.auth(TMP())
-    assert TMP.headers['Authorization'] == 'Bearer a'
-
-    # custom_token_server with its own auth class
-    responses.add(responses.POST, 'https://example.com', json={'token': 'a'})
-    session = Auth(type='custom_token_server',
-                   args=['POST', 'https://example.com'],
-                   kwargs={'auth': {'type': 'basic', 'args': ['u', 'p']}}).get_session()
-
-    session.auth(TMP())
-    assert responses.calls[1].request.headers['Authorization'].startswith('Basic')
+        ({'$match': {'$and': [
+            {'domain': 'truc', 'indic0': '{{my_indic[0]}}'},
+            {'indic1': '{{my_indic[1]}}', 'indic2': 'yo_{{my_indic[2]}}'},
+            {'indic_list': '{{my_indic}}'}
+         ]}},
+         {'my_indic': ['0', 1, '2']},
+         {'$match': {'$and': [
+             {'domain': 'truc', 'indic0': '0'},
+             {'indic1': 1, 'indic2': 'yo_2'},
+             {'indic_list': ['0', 1, '2']}
+         ]}}),
+
+        ({'$match': {'$and': [
+            {'domain': 'truc', 'indic0': '%(my_indic_0)s'},
+            {'indic1': '%(my_indic_1)s', 'indic2': 'yo_%(my_indic_2)s'},
+            {'indic_list': '%(my_indic)s'}
+         ]}},
+         {'my_indic_0': '0', 'my_indic_1': 1, 'my_indic_2': '2', 'my_indic': ['0', 1, '2']},
+         {'$match': {'$and': [
+             {'domain': 'truc', 'indic0': '0'},
+             {'indic1': 1, 'indic2': 'yo_2'},
+             {'indic_list': ['0', 1, '2']}
+         ]}}),
+
+        ({'$match': {'$and': [
+            {'domain': 'truc', 'indic0': '{{my_indic["zero"]}}'},
+            {'indic1': '{{my_indic["one"]}}', 'indic2': 'yo_{{my_indic["two"]}}'},
+            {'indic_list': '{{my_indic}}'}
+        ]}},
+         {'my_indic': {'zero': '0', 'one': 1, 'two': '2'}},
+         {'$match': {'$and': [
+             {'domain': 'truc', 'indic0': '0'},
+             {'indic1': 1, 'indic2': 'yo_2'},
+             {'indic_list': {'zero': '0', 'one': 1, 'two': '2'}}
+         ]}})
+    ]
+    for test in tests:
+        assert nosql_apply_parameters_to_query(test[0], test[1]) == test[2]
```

Comment:
maybe name the variables rather that using the index in the tuple to make it more readable